scripts/test/Muon/fitting_widgets/model_fitting/model_fitting_model_test.py

```diff
@@ -0,0 +1,180 @@
+        self.assertEqual(self.model.parameter_combination_workspace_name("A0", "A1"), None)
+
+    def test_that_parameter_combination_group_name_returns_the_expected_name_for_the_provided_parameters(self):
+        self.model.result_table_names = self.result_table_names
+        self.assertEqual(self.model.parameter_combination_group_name(), "Result1; Parameter Combinations")
+
+    def test_that_parameter_combination_group_name_returns_none_when_there_are_no_results_tables(self):
+        self.assertEqual(self.model.parameter_combination_group_name(), None)
+
+    def test_that_get_workspace_names_to_display_from_context_only_returns_the_names_that_exist_in_the_ADS(self):
+        self.model.context.results_context.result_table_names = self.result_table_names
+
+        table_workspace = WorkspaceFactory.createTable()
+        add_ws_to_ads("Result1", table_workspace)
+
+        self.assertEqual(self.model.get_workspace_names_to_display_from_context(), ["Result1"])
+
+    def test_that_create_x_and_y_parameter_combination_workspaces_will_create_the_expected_parameter_data(self):
+        self.model.result_table_names = self.result_table_names
+        self.model.current_result_table_index = 0
+
+        table = create_results_table()
+        add_ws_to_ads("Result1", table)
+
+        x_parameters, y_parameters = self.model.create_x_and_y_parameter_combination_workspaces()
+
+        self.assertEqual(list(x_parameters), ["workspace_name", "A0", "A1"])
+        self.assertEqual(list(y_parameters), ["workspace_name", "A0", "A1"])
+
+        self.assertAlmostEqual(self.model.fitting_context.x_parameters["A0"][0], 0.1, delta=0.000001)
+        self.assertAlmostEqual(self.model.fitting_context.x_parameters["A0"][1], 0.3, delta=0.000001)
+        self.assertAlmostEqual(self.model.fitting_context.x_parameters["A0"][2], 0.5, delta=0.000001)
+        self.assertAlmostEqual(self.model.fitting_context.y_parameters["A1"][0], 0.2, delta=0.000001)
+        self.assertAlmostEqual(self.model.fitting_context.y_parameters["A1"][1], 0.4, delta=0.000001)
+        self.assertAlmostEqual(self.model.fitting_context.y_parameters["A1"][2], 0.6, delta=0.000001)
+
+        self.assertEqual(self.model.fitting_context.y_parameters["workspace_name"],
+                         ["MUSR62260; Group; bottom; Asymmetry; MA", "MUSR62260; Group; top; Asymmetry; MA",
+                          "MUSR62260; Group; fwd; Asymmetry; MA"])
+
+    def test_that_create_x_and_y_parameter_combination_workspaces_will_create_the_expected_parameter_workspaces(self):
+        self.model.result_table_names = self.result_table_names
+        self.model.current_result_table_index = 0
+
+        table = create_results_table()
+        add_ws_to_ads("Result1", table)
+
+        self.model.create_x_and_y_parameter_combination_workspaces()
+
+        self.assertTrue(check_if_workspace_exist("Result1; Parameter Combinations"))
```

Comment:
could we group all of the "x vs x" tests together? Just to make it a bit easier to understand thats why they are expected to be False.